### PR TITLE
Fix #1009: Apply DesignModules to the main frame of MediaCompose

### LINKED ISSUE #1009: Use @DesignModules in MediaCompose reference app

### PR DESCRIPTION
**Pull request by yiqunw700**

### CODE REVIEW

**Review by timothyfroehlich — APPROVED** [review]

**Review by rylin8 — CHANGES_REQUESTED** [review]
The fact that this compiles and runs is great, and proves that design modules are working as intended. However, one thing I forgot about it that we also use the annotated code to generate a JSON file for the list widget. All of the @DesignPreviewContent and @DesignContent annotations are used purely for generation of a JSON file that is uploaded into the list widget so that the widget knows what nodes can go inside the widget, and what preview content to show. Generation of this JSON file with design modules has not been tested, and my guess is that it will not work, so please don't merge this yet. Since this is not urgent, I'd suggest waiting until the JSON generation is fixed before merging this.

**Comment by rylin8** [comment]
Filed https://github.com/google/automotive-design-compose/issues/982

File: reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/Modules.kt
Comment (rylin8) [inline reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/Modules.kt:43]: Since this is a nested module, you need to use `@DesignModuleProperty`. It's annoying that KSP doesn't complain that `@DesignModule` is applied to the wrong type, since it is an annotation that targets function parameters, not class properties. I'm hoping KSP fixes the issue where I need to use a different annotation for parameters/properties.

**Comment by rylin8** [comment]
Also I'd run `./gradlew spotlessApply` to format the kotlin files. @timothyfroehlich do we not do a lint check anymore?

**Comment by rylin8** [comment]
Code to fix JSON generation: https://github.com/google/automotive-design-compose/pull/1004

File: reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/Modules.kt
Comment (yiqunw700) [reply reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/Modules.kt]: Done.

**Comment by rylin8** [comment]
I'm confused how this is building when my change has not merged https://github.com/google/automotive-design-compose/pull/1004

**Comment by yiqunw700** [comment]
> I'm confused how this is building when my change has not merged #1004

It doesn't. I cherrypicked your change locally and made the change on top of that. Still need to wait for your change to get merged.

**Comment by rylin8** [comment]
Ok my change merged, could you see if the failure is fixed now?

**Review by rylin8 — APPROVED** [review]

**Pull request merged**
